### PR TITLE
Merge 1.16

### PR DESCRIPTION
**Pull request by jepler**
Notable changes for CircuitPython:
 * In the Python core, OSError exceptions now support the ".errno" attribute
 * There has been a minor breaking change to a relative import exception: what was previously a ValueError was changed to ImportError, following the same change in CPython

### CODE REVIEW

**Comment by jepler** [comment]
Unix tests pass locally, I'll let CI chew on the rest and see where we're at.

File: tools/pyboard.py [hunk 1]
```diff
@@ -1,11 +1,36 @@
-#!/usr/bin/env python
-
-# SPDX-FileCopyrightText: Copyright (c) 2014-2019 Damien P. George
+#!/usr/bin/env python3
+#
+# SPDX-FileCopyrightText: Copyright (c) 2014-2016 Damien P. George
```
Comment (tannewt) [inline tools/pyboard.py:3]:
```suggestion
# SPDX-FileCopyrightText: Copyright (c) 2014-2021 Damien P. George
```

And delete the chunk below.

**Review by tannewt — CHANGES_REQUESTED** [review]
Thanks for taking on this merge!

Please delete `ports/zephyr`, `ports/rp2`, `ports/esp32` and `ports/esp8266`. We have our own ports and the extra folders confuse folks.

I delete `tests/multi_{net,bluetooth}` too since we don't use it.

We may want to delete `tools/mpremote` as well. I don't think it'll work with CircuitPython and I doubt we want to support it.

**Comment by jepler** [comment]
@tannewt OK, got those all I think.  Do you want me to try to "squash" this to a single merge commit, or did your recent experiences lead you to think that's not such a good idea?

**Comment by tannewt** [comment]
No need to squash. It's funny with merge commits.

I see a `ports/qemu-arm` left that I forgot to list. For reference, I used this script to delete other ports: https://github.com/adafruit/circuitpython/blob/main/tools/merge_micropython.py#L13

**Review by tannewt — APPROVED** [review]
Looks great! Thank you!

**Comment by ladyada** [comment]
woo hoo!

**Pull request merged**